compiler/src/x86_arch_full.ml

```diff
@@ -47,6 +48,140 @@ module X86_core = struct
     | DIV _ | IDIV _ -> false
     | _ -> true
 
+  let is_doit_asm_op (o : asm_op) =
```

Comment:
This function should not use | _ -> false.
All the cases should be explicite. Else in the future 
if we add new instruction the Ocaml compiler will not 
complains that we have not added a case. 
Also why this function is not automatically generated ?

Reply:
I will ditch the `_ -> false` and make it generate the false cases.

The function is semi-automatically generated. That is what the `extract_doit.py` script does. It goes like 95% of the way. The remaining 5% is very tricky to automate. Jasmin is not fully consistent in how it includes instructions and when it has all or only some of their variants. Also for example `MOV` or `PUSH` and `POP` do not have all of their opcodes on the DOIT list. The opcodes that manipulate (push or pop) the segment registers are non-DOIT, but those will never be used by Jasmin so it is fine. But essentially things like this need to be argued when constructing the list. Or for example the naming `MULx_lo_hi` or the fact that `CMOVcc` doesn't have the condition info in the asm_op type, etc. I could spend more time making it fully automated but it would just be full of manual special cases anyway and thus not future proof, which is like the whole point of automating it.

Reply:
This now makes all cases explicit.